genesyscloud/resource_genesyscloud_journey_segment.go

```diff
@@ -0,0 +1,623 @@
+		}
+
+		pageCount = *journeySegments.PageCount
+	}
+
+	return resources, nil
+}
+
+func journeySegmentExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllJourneySegments),
+		RefAttrs:         map[string]*RefAttrSettings{}, // No references
+	}
+}
+
+func resourceJourneySegment() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Genesys Cloud Journey Segment",
+		CreateContext: createWithPooledClient(createJourneySegment),
+		ReadContext:   readWithPooledClient(readJourneySegment),
+		UpdateContext: updateWithPooledClient(updateJourneySegment),
+		DeleteContext: deleteWithPooledClient(deleteJourneySegment),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema:        journeySegmentSchema,
+	}
+}
+
+func createJourneySegment(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	journeyApi := platformclientv2.NewJourneyApiWithConfig(sdkConfig)
+	journeySegment := buildSdkJourneySegment(d)
+
+	log.Printf("Creating journey segment %s", *journeySegment.DisplayName)
+	result, resp, err := journeyApi.PostJourneySegments(*journeySegment)
+	if err != nil {
+		return diag.Errorf("failed to create journey segment %s: %s\n(input: %+v)\n(resp: %s)", *journeySegment.DisplayName, err, *journeySegment, resp.RawBody)
+	}
+
+	d.SetId(*result.Id)
+
+	log.Printf("Created journey segment %s %s", *result.DisplayName, *result.Id)
+	return readJourneySegment(ctx, d, meta)
+}
+
+func readJourneySegment(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	journeyApi := platformclientv2.NewJourneyApiWithConfig(sdkConfig)
+
+	log.Printf("Reading journey segment %s", d.Id())
+	return withRetriesForRead(ctx, d, func() *resource.RetryError {
+		journeySegment, resp, getErr := journeyApi.GetJourneySegment(d.Id())
+		if getErr != nil {
+			if isStatus404(resp) {
+				return resource.RetryableError(fmt.Errorf("failed to read journey segment %s: %s", d.Id(), getErr))
+			}
+			return resource.NonRetryableError(fmt.Errorf("failed to read journey segment %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, resourceJourneySegment())
+		flattenJourneySegment(d, journeySegment)
+
+		log.Printf("Read journey segment %s %s", d.Id(), *journeySegment.DisplayName)
+		return cc.CheckState()
+	})
+}
+
+func updateJourneySegment(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	journeyApi := platformclientv2.NewJourneyApiWithConfig(sdkConfig)
+	journeySegment := buildSdkPatchSegment(d)
+
+	log.Printf("Updating journey segment %s", d.Id())
+	if _, resp, err := journeyApi.PatchJourneySegment(d.Id(), *journeySegment); err != nil {
```

Comment:
Have a search for `retryWhen(isVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics)` in other resources. We wrap updates in that retry logic because sometimes there's a version mismatch when updating resources